src/Catalog/Extensions/DbDataReaderExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace System.Data.Common
+{
+    public static class DbDataReaderExtensions
```

Comment:
Two methods here are almost identical, how about refactoring it to not have copy-pasted code?
I would suggest adding `private static T ReadColumn<T>(DbDataReader dr, string cn, Func<int, T> provideValue)` and make other use it supplying type-specific `provideValue` delegates.

Reply:
I managed to reduce code, but the tricky part is that the generic constraints for value types and reference types are contradictory. Managed to work around that though using the `dynamic` keyword :) Please have another look. I could duplicate code again by making a proper value type and reference type variant of the method.

Reply:
Why does it need a `return` type of `dynamic` and not just `T`?

Reply:
Actually, that is possible now since I'm passing in the `nullValue` as an argument.
Updated https://github.com/NuGet/NuGet.Services.Metadata/pull/542/commits/8b7ab6b8a064942a58efd99a811fa889445dbb63